servicenow/stackstate_checks/servicenow/servicenow.py

```diff
@@ -211,10 +217,10 @@ def _process_component_relations(self, instance_config, batch_size, timeout, rel
 
         completed = False
         while not completed:
-            state = self._collect_component_relations(instance_config, timeout, offset, batch_size)
-
-            if "result" in state:
-                for relation in state.get('result', []):
+            relations = self._collect_component_relations(instance_config, timeout, offset, batch_size)
+            total_relations = len(relations.get("result"))
+            if "result" in relations and isinstance(relations["result"], list):
```

Comment:
Everything in this loop except the processing of the component/relation is duplicate logic. My advise, extract this code into a higher order function.

Reply:
Extracted collect and process to higher function.